components/renku_data_services/project/core.py

```diff
@@ -15,3 +22,54 @@ def validate_project_patch(patch: apispec.ProjectPatch) -> models.ProjectPatch:
         description=patch.description,
         keywords=keywords,
     )
+
+
+async def copy_project(
+    project_id: ULID,
+    user: APIUser,
+    name: str,
+    namespace: str,
+    slug: str | None,
+    description: str | None,
+    repositories: list[models.Repository] | None,
+    visibility: str | None,
```

Comment:
nitpick: you should already pass in the `Visibility | None` object here, not str. Blueprint should be dealing with parsing input.